code/modules/mining/ores_coins.dm

```diff
@@ -369,6 +370,55 @@
 /obj/item/weapon/coin/is_screwdriver(var/mob/user)
 	return user.a_intent == I_HURT
 
+/obj/item/weapon/coin/proc/coinflip(mob/user as mob, thrown, rigged = FALSE)
+	var/matrix/flipit = matrix()
+	flipit.Scale(0.2,1)
+	animate(src, transform = flipit, time = 2, easing = QUAD_EASING)
+	flipit.Scale(5,1)
+	flipit.Invert()
+	flipit.Turn(rand(1,359))
+	animate(src, transform = flipit, time = 2, easing = QUAD_EASING)
+	flipit.Scale(0.2,1)
+	animate(src, transform = flipit, time = 2, easing = QUAD_EASING)
+	if (pick(0,1))
+		sideup = "heads-up."
+		flipit.Scale(5,1)
+		flipit.Turn(rand(1,359))
+		animate(src, transform = flipit, time = 2, easing = QUAD_EASING)
+	else
+		sideup = "tails-up."
+		flipit.Scale(5,1)
+		flipit.Invert()
+		flipit.Turn(rand(1,359))
+		animate(src, transform = flipit, time = 2, easing = QUAD_EASING)
+	if (prob(0.1) || rigged)
+		flipit.Scale(0.2,1)
+		animate(src, transform = flipit, time = 2, easing = QUAD_EASING)
+		sideup = "on the side!"
+	if(!thrown)
+		user.visible_message("<span class='notice'>[user] flips [src]. It lands [sideup]</span>", \
+							 "<span class='notice'>You flip [src]. It lands [sideup]</span>", \
+							 "<span class='notice'>You hear [src] landing.</span>")
+	else 
+		if(!throwing) //coin was thrown and is coming to rest
+			visible_message("<span class='notice'>[src] stops spinning, landing [sideup]</span>")
+	
+/obj/item/weapon/coin/examine(var/mob/user)
+	..()
+	to_chat(user, "<span class='notice'>[src] is [sideup]</span>")
```

Comment:
Please put the period here rather than in the variable.

Reply:
No, because one of them has a `!` and I'm not going to replace a single character with four lambdas.